dll/win32/shell32/shellrecyclebin/recyclebin_v5_enumerator.c

```diff
@@ -17,6 +18,12 @@ struct RecycleBin5File
     WCHAR FullName[ANY_SIZE];
 };
 
+EXTERN_C HRESULT WINAPI SHCreateFileExtractIconW(LPCWSTR pszPath, DWORD dwFileAttributes, REFIID riid, void **ppv);
```

Comment:
Could you add this to `shlobj.h` instead please (along with the -A version, ideally)?

Reply:
well yeah, i can add it, but other parts of the ROS uses this format to import this function

Reply:
oh, yeah, i know what you mean, i can fix it

Reply:
and what's more SHCreateFileExtractIconA it's not implemented at all, many parts of the ROS says that we should fix the headers and idk what to touch to get it correctly if i'm honest

Reply:
If you think it's going out of the PR's topic I can manage the implementation of SHCreateFileExtractIconA and moving the prototypes to shlobj.h .

Reply:
I can help with that.

Reply:
i can do it, that's not the problem, it's in the topic, the only thing idk it's, should i copy the function and replace the function in -W with functions that use -A or convert -A to -W and the use the function?

Reply:
Just add these two lines to `shlobj.h`, that's all:
```
EXTERN_C HRESULT WINAPI SHCreateFileExtractIconA(LPCSTR pszPath, DWORD dwFileAttributes, REFIID riid, void **ppv);
EXTERN_C HRESULT WINAPI SHCreateFileExtractIconW(LPCWSTR pszPath, DWORD dwFileAttributes, REFIID riid, void **ppv);
```

Reply:
By the way, from Geoff:
https://www.geoffchappell.com/studies/windows/shell/shell32/api/index.htm
```
SHCreateFileExtractIconW (743) | 6.0 and higher | documented for settlement
```
and there is no A api being exported.

Reply:
Done